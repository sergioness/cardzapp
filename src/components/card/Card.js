import React from 'react';

class Card extends React.PureComponent {
    render() {
        const {id, name, email} = this.props.info;
        return (
            <div className='tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
                <img alt={`${name} thumbnail`} src={`https://robohash.org/${id}`}/>
                <h2>{name}</h2>
                <div>{email}</div>
            </div>
        );
    }
}
export default Card;