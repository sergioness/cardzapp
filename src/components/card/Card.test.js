import Enzyme from 'enzyme';
import React from 'react';
import Card from './Card';
import toJson from 'enzyme-to-json';

describe('Component \'Card\'', () => {
  it('shallow matches the snapshot', () => {
    const mockCard = {
      id: 0,
      name: 'dummy card',
      email: 'card@card.com'
    };
    const shallowCard = Enzyme.shallow(<Card info={mockCard}/>);
    expect(toJson(shallowCard, {mode: 'deep'})).toMatchSnapshot();
  });
});
