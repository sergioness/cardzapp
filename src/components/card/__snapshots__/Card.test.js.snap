// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component 'Card' shallow matches the snapshot 1`] = `
<div
  className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="dummy card thumbnail"
    src="https://robohash.org/0"
  />
  <h2>
    dummy card
  </h2>
  <div>
    card@card.com
  </div>
</div>
`;
